coderd/httpmw/logger_internal_test.go

```diff
@@ -0,0 +1,154 @@
+package httpmw
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+	"time"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/v2/coderd/tracing"
+	"github.com/coder/coder/v2/testutil"
+	"github.com/coder/websocket"
+)
+
+func TestRequestLogger_WriteLog(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	sink := &fakeSink{}
+	logger := slog.Make(sink)
+	logger = logger.Leveled(slog.LevelDebug)
+	logCtx := NewRequestLogger(logger, "GET", time.Now())
+
+	// Add custom fields
+	logCtx.WithFields(
+		slog.F("custom_field", "custom_value"),
+	)
+
+	// Write log for 200 status
+	logCtx.WriteLog(ctx, http.StatusOK)
+
+	if len(sink.entries) != 1 {
+		t.Fatalf("expected 1 log entry, got %d", len(sink.entries))
+	}
+
+	if sink.entries[0].Message != "GET" {
+		t.Errorf("expected log message to be 'GET', got '%s'", sink.entries[0].Message)
+	}
+
+	if sink.entries[0].Fields[0].Value != "custom_value" {
+		t.Errorf("expected a custom_field with value custom_value, got '%s'", sink.entries[0].Fields[0].Value)
+	}
```

Comment:
You should probably check the status field too. I would use a loop to first read all of the fields into a `map[string]Field` and then check all four (and ensure no extra fields got added either by doing a Len check)